### PR TITLE
Support imported targets in FindIgnOGRE.cmake

### PR DESCRIPTION
**Pull request by peci1**
# Bug Report

## Summary
Currently, if FindOGRE.cmake returns an imported target (like `OGRE::OgreMain`), the `FindIgnOGRE` module appends a `.lib` suffix to it, which is obviously wrong.

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [x] `codecheck` passed (See
  [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See
  [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review
[another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+archived%3Afalse+)
to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge**

### CODE REVIEW

File: cmake/FindIgnOGRE.cmake [hunk 1]
```diff
@@ -191,7 +191,9 @@ else()
         # Some Ogre libraries are not using the .lib extension
         set(postfix "")
         if(NOT ogre_lib MATCHES ".lib$")
-          set(postfix ".lib")
+          if(NOT ogre_lib MATCHES "::")
```
Comment (j-rivero) [inline cmake/FindIgnOGRE.cmake:194]:
```suggestion
          # Do not consider imported targets as libraries
          if(NOT ogre_lib MATCHES "::")
```

**Review by j-rivero — APPROVED** [review]
Looks good to me. One minor suggestion to comment the change and we are good to go.

**Comment by peci1** [comment]
Thanks, comment added.

**Comment by adlarkin** [comment]
There was a warning on Homebrew CI that should have been fixed by https://github.com/ignition-tooling/release-tools/pull/401. I have re-triggered Homebrew CI to make sure it comes back green.

**Review by adlarkin — APPROVED** [review]

**Pull request merged**